roboteam_ai/src/conditions/BallInDefenseAreaAndStill.cpp

```diff
@@ -13,14 +13,14 @@ namespace ai {
 BallInDefenseAreaAndStill::BallInDefenseAreaAndStill(std::string name, bt::Blackboard::Ptr blackboard)
         :Condition(std::move(name), std::move(blackboard)) { };
 
-void BallInDefenseAreaAndStill::initialize() {
+void BallInDefenseAreaAndStill::onInitialize() {
     if (properties->hasBool("theirDefenceArea")) {
         theirDefenceArea = properties->getBool("theirDefenceArea");
     }
     else {theirDefenceArea=false;}
 }
 
-bt::Node::Status BallInDefenseAreaAndStill::update() {
+bt::Node::Status BallInDefenseAreaAndStill::onUpdate() {
     Vector2 ballPos;
     auto ball=World::getBall();
```

Comment:
the ball object is already available here so don't need to get it

Reply:
Yes and no,

Currently the implementation in Condition.h is: *if* there is a "ROLE" in properties, get the robot corresponding to that role, and get the ball. 
No ball or no robot? Then return;

If the Condition is used in a non-role scenario though, (which is possible for Conditions, not for Skills) you might want to get the ball in the condition itselves at 'onUpdate' if needed, as it is not done in 'update'.